andela/mrm_api
# #164167409 (V2)  Enable admin to query feedback responses based on date range

#### What does this PR do?
Enable admin to query feedback responses based on date range

#### Description of Task to be completed?
- Currently, when querying responses, one can not query responses by date. This PR allows an admin to query responses by date range. 
- The query can also be run with a `room`  as shown below
- The code has been refactored to cater for both `date range` and `count range` since the functionality is almost similar

#### How should this be manually tested?
- Pull this branch `ft-query-responses-with-data-range-164167409`
- Run the following query
```
query{
    allRoomResponses(lowerDateLimit: "2019 feb 20", upperDateLimit: "2019 feb 25" ){
        responses{
            totalResponses
            roomName
            response{
                responseId
                missingItems
            }
        }
    }
}
```
or 

```
query{
    allRoomResponses(lowerDateLimit: "2019 feb 20", upperDateLimit: "2019 feb 25", room: "Entebbe" ){
        responses{
            totalResponses
            roomName
            response{
                responseId
                missingItems
            }
        }
    }
}
```
You should be able to query all the responses that fall between that date range.

#### Any background context you want to provide?
This is an extension to query all responses.

#### What are the relevant pivotal tracker stories?
[#164167409](https://www.pivotaltracker.com/story/show/164167409)

#### Screenshots (if appropriate)
**Before**
![screen shot 2019-02-28 at 6 44 21 am](https://user-images.githubusercontent.com/22786444/53539960-ae430500-3b24-11e9-92b0-178fd3fc0437.png)

**After**
![screen shot 2019-02-28 at 6 45 15 am](https://user-images.githubusercontent.com/22786444/53539971-b733d680-3b24-11e9-8ac3-cf1ebeb5a8ba.png)
 
This screenshot shows date range for a specific room
![screen shot 2019-02-28 at 8 35 53 pm](https://user-images.githubusercontent.com/22786444/53595652-dc6d2700-3bae-11e9-988a-49d011f25adb.png)
